subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ZipHasher.java

```diff
@@ -53,12 +63,28 @@ public static boolean isZipFile(final String name) {
         return false;
     }
 
+    public static boolean isManifestFile(final String name) {
+        return name.equals("META-INF/MANIFEST.MF");
+    }
+
+    public static boolean isManifestPropertyFile(final String name) {
+        return name.startsWith("META-INF") && name.endsWith(".properties");
+    }
+
     private final ResourceHasher resourceHasher;
     private final ResourceFilter resourceFilter;
+    private final ResourceFilter attributeResourceFilter;
+    private final ResourceFilter propertyResourceFilter;
```

Comment:
Those are `ByteContentHasher` because manifests and properties files aren't just filtered, but normalized regardless of the configured filters. `MapEntry` exists for sorting primarily and IIRC there were duplicate key cases we need to avoid so sorted maps were a bad fit.

Agree it's shoehorning a little here, but `ResourceFilter`/`PathMatcher` felt reasonably general purpose, in that it's simple matching against one or more string parts. Per my [other comment](https://github.com/gradle/gradle/pull/11937#discussion_r420993610) I think we want at least simple wildcard matches. `ResourceEntryFilter` is a good name, I'll start there.